cpp/models/lct_secir/initializer_flows.h

```diff
@@ -0,0 +1,251 @@
+                          "are needed but "
+                          "just {} are given.",
+                          -init[i], m_flows.get_num_time_points());
+                return true;
+            }
+        }
+
+        // Update initial value vector of the mpdel.
+        m_model.set_initial_values(init);
+
+        // Check if constraints were fulfilled.
+        return check_constraints();
+    }
+
+    /**
+     * @brief Setter for the tolerance used to calculate the maximum support of ErlangDensity%s.
+     *
+     * @param[in] new_tol New tolerance.
+     */
+    void set_tol_for_support_max(ScalarType new_tol)
+    {
+        m_tol = new_tol;
+    }
+
+private:
+    /**
+     * @brief Checks constraints of the Initializer inclusive check for the model.
+     * @return Returns true if one (or more) constraint(s) are not satisfied, otherwise false. 
+     */
+    bool check_constraints() const
+    {
+        if (!(Eigen::Index(InfectionTransition::Count) == m_flows.get_num_elements())) {
+            log_error("Initial condition size does not match subcompartments.");
+            return true;
+        }
+
+        if (!(floating_point_equal(0., m_flows.get_last_time(), 1e-8, 1e-14))) {
+            log_error("Last time point in flows has to be 0.");
+            return true;
+        }
+
+        for (int i = 1; i < m_flows.get_num_time_points(); i++) {
+            if (!(floating_point_equal(m_dt, m_flows.get_time(i) - m_flows.get_time(i - 1), 1e-8, 1e-14))) {
+                log_error("Time points in flows have to be equidistant.");
+                return true;
+            }
+        }
+
+        if (!(m_dt < 1)) {
+            log_warning("Step size was set very large. The result could be distorted.");
+            return true;
+        }
+
+        // Check if model is valid and the calculated initial value vector was valid.
+        return m_model.check_constraints();
+    }
+
+    /**
+     * @brief Computes a vector with the number of people in each subcompartment for one InfectionState.
+     *
+     * With this function, partial result of compute_initialization_vector are achieved.
+     *
+     * @param[in] idx_incoming_flow Index of the flow which is relevant for the calculation, so the flow to the InfectionState.
+     * @param[in] transition_rate Specifies the transition rate of the InfectionState. Is equal to 1 / (expected Time in the InfectionState).
+     * @tparam State The InfectionState for which the partial result should be calculated.
+     * @return Vector with a possible initialization for the subcompartments of the InfectionState State. 
+     *      Subcompartment is set to -1 if calculation was not possible.
+     */
+    template <InfectionState State>
+    Eigen::VectorXd compute_compartment(Eigen::Index idx_incoming_flow, ScalarType transition_rate) const
+    {
+        int num_infectionstates = LctState::template get_num_subcompartments<State>();
```

Comment:
```suggestion
        int num_subcompartments = LctState::template get_num_subcompartments<State>();
```
I find num_infectionstates confusing here.